intermediate_source/dynamic_quantization_bert_tutorial.rst

```diff
@@ -0,0 +1,550 @@
+in PyTorch `here <https://github.com/pytorch/pytorch/#installation>`_ and HuggingFace Github Repo `here <https://github.com/huggingface/transformers#installation>`_.
+In addition, we also install `scikit-learn <https://github.com/scikit-learn/scikit-learn>`_ package, as we will reuse its
+built-in F1 score calculation helper function.
+
+.. code:: shell
+
+   pip install sklearn
+   pip install transformers
+
+
+Because we will be using the experimental parts of the PyTorch, it is
+recommended to install the latest version of torch and torchvision. You
+can find the most recent instructions on local installation `here
+<https://pytorch.org/get-started/locally/>`_. For example, to install on
+Mac:
+
+.. code:: shell
+
+   yes y | pip uninstall torch tochvision
+   yes y | pip install --pre torch -f https://download.pytorch.org/whl/nightly/cu101/torch_nightly.html
+
+
+
+
+1.2 Import the necessary modules
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this step we import the necessary Python modules for the tutorial.
+
+.. code:: python
+
+    from __future__ import absolute_import, division, print_function
+
+    import logging
+    import numpy as np
+    import os
+    import random
+    import sys
+    import time
+    import torch
+
+    from argparse import Namespace
+    from torch.utils.data import (DataLoader, RandomSampler, SequentialSampler,
+                                  TensorDataset)
+    from tqdm import tqdm
+    from transformers import (BertConfig, BertForSequenceClassification, BertTokenizer,)
+    from transformers import glue_compute_metrics as compute_metrics
+    from transformers import glue_output_modes as output_modes
+    from transformers import glue_processors as processors
+    from transformers import glue_convert_examples_to_features as convert_examples_to_features
+
+    # Setup logging
+    logger = logging.getLogger(__name__)
+    logging.basicConfig(format = '%(asctime)s - %(levelname)s - %(name)s -   %(message)s',
+                        datefmt = '%m/%d/%Y %H:%M:%S',
+                        level = logging.WARN)
+
+    logging.getLogger("transformers.modeling_utils").setLevel(
+       logging.WARN)  # Reduce logging
+
+    print(torch.__version__)
+
+We set the number of threads to compare the single thread performance between FP32 and INT8 performance.
+In the end of the tutorial, the user can set other number of threads by building PyTorch with right parallel backend.
+
+.. code:: python
+    torch.set_num_threads(1)
+    print(torch.__config__.parallel_info())
+
+
+1.3 Download the dataset
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before running MRPC tasks we download the `GLUE data
+<https://gluebenchmark.com/tasks>`_ by running `this script
+<https://gist.github.com/W4ngatang/60c2bdb54d156a41194446737ce03e2e>`_
+and unpack it to a directory ``glue_data``.
+
+
+.. code:: shell
+
+   python download_glue_data.py --data_dir='glue_data' --tasks='MRPC'
+
+
+1.4 Learn about helper functions
```

Comment:
Suggest merging this section with 1.2 as we are explaining what we are doing in 1.2 here, suggested change is shown above.